Release notes for v2.0.0-rc1 (#5868)
grafana/k6: release notes/v2.0.0-rc1.md

**Migration:** Remove any `K6_OTEL_SINGLE_COUNTER_FOR_RATE=true` configuration. If you were using the old pair-of-counters format, update your dashboards and queries to use the single counter with `condition` attribute instead.

### Removed `K6_BINARY_PROVISIONING` environment variable [#5734](https://github.com/grafana/k6/pull/5734)

The `K6_BINARY_PROVISIONING` environment variable, deprecated in v1.2.0, has been removed.

**Migration:** Remove `K6_BINARY_PROVISIONING` from your environment. Auto-extension-resolution is enabled by default; `K6_AUTO_EXTENSION_RESOLUTION` only needs to be set explicitly if you want to disable it.

### Removed `K6_ENABLE_COMMUNITY_EXTENSIONS` environment variable [#5733](https://github.com/grafana/k6/pull/5733)

The `K6_ENABLE_COMMUNITY_EXTENSIONS` environment variable has been removed. The community and cloud extension catalogs were merged server-side, making this flag a no-op since the catalogs were unified.

**Migration:** Remove `K6_ENABLE_COMMUNITY_EXTENSIONS` from your environment. Community extensions are now resolved through the default build service URL automatically.

> ### Stack ID is now required for all `k6 cloud` commands [#5833](https://github.com/grafana/k6/pull/5833)

Reviewer: ```suggestion
### Stack is now required for all `k6 cloud` commands [#5833](https://github.com/grafana/k6/pull/5833)
```
this should be the URL or the Slug. The ID is mostly an additional option that I'd prefer to not advertise for.

> Providing a stack is now mandatory for all `k6 cloud` commands (`k6 cloud run`, `k6 cloud upload`, `k6 cloud run --local-execution`). Previously, omitting a stack would fall back to the first available stack with a deprecation warning — that fallback has been removed. Likewise, `k6 cloud login` now requires both a token and a stack; passing one without the other fails with an explicit error. The legacy v1 token-only validation path (`/v1/validate-token`) has also been removed.

Reviewer: ```suggestion
Providing a stack is now mandatory for all `k6 cloud` commands (`k6 cloud run`, `k6 cloud upload`, `k6 cloud run --local-execution`). Previously, omitting a stack would fall back to the first available stack with a deprecation warning — that fallback has been removed. Likewise, `k6 cloud login` now requires both a token and a stack; passing one without the other fails with an explicit error.
```
This sounds a detail in this context. I'd remain focus on the impact for the user.

> **Migration:** Ensure a stack is always provided via `k6 cloud login --stack`, the `K6_CLOUD_STACK_ID` environment variable, or the `stackID` script option before running any `k6 cloud` command.

Reviewer: ```suggestion
**Migration:** Ensure a stack is always provided via `k6 cloud login --stack` passing the stack slug or the full URL. Alternatively, the `K6_CLOUD_STACK_ID` environment variable, or the `stackID` script option are available to be set before running any `k6 cloud` command .
```

- InfluxDB authentication is no longer configurable via a login command. Use environment variables such as `K6_INFLUXDB_*` to configure the InfluxDB output directly.

### `k6 cloud script.js` no longer runs a test [#5624](https://github.com/grafana/k6/pull/5624)

Running `k6 cloud` without an explicit subcommand (i.e., the old positional-argument form `k6 cloud script.js`) now shows the help output instead of attempting to run the test. The `run` subcommand was already the recommended path since `k6 cloud run` was introduced.

**Migration:** Replace `k6 cloud script.js` with `k6 cloud run script.js`.

### Removed `--upload-only` flag [#5844](https://github.com/grafana/k6/pull/5844)

The `--upload-only` flag on the `k6 cloud` command has been removed.

**Migration:** Use `k6 cloud upload script.js` to upload a test without running it.

### Removed `--no-summary` flag [#5729](https://github.com/grafana/k6/pull/5729)

The `--no-summary` flag has been removed.

**Migration:** Replace `--no-summary` with `--summary-mode=disabled`.

### Removed `--summary-mode=legacy` [#5730](https://github.com/grafana/k6/pull/5730)

The `legacy` value for `--summary-mode` has been removed.

> **Migration:** Replace `--summary-mode=legacy` with `--summary-mode=compact` (the default) or `--summary-mode=full` for detailed output.

Reviewer 2: I'd argue there's no real migration, the `legacy` mode has been dropped and that's it. The other two are alternatives, other variants, but not a real replacement in the sense of another way of getting _exactly_ the same thing.

Author: Do you recommend that we do not have this under "migration" but .. a replacement or ?

Reviewer 2: Either completely dropping the "migration" section, or adding a short sentence explaining that there's other two modes available, perhaps inviting the reader to go to the docs and see more there: https://grafana.com/docs/k6/latest/results-output/end-of-test/#summary-modes

Author: [7a0859d](https://github.com/grafana/k6/pull/5868/commits/7a0859de650743c14291b24848ca583ca36ca6ce)

The `K6_ENABLE_COMMUNITY_EXTENSIONS` environment variable has been removed. The community and cloud extension catalogs were merged server-side, making this flag a no-op since the catalogs were unified.

**Migration:** Remove `K6_ENABLE_COMMUNITY_EXTENSIONS` from your environment. Community extensions are now resolved through the default build service URL automatically.

### Stack is now required for all `k6 cloud` commands [#5833](https://github.com/grafana/k6/pull/5833)

Providing a stack is now mandatory for all `k6 cloud` commands (`k6 cloud run`, `k6 cloud upload`, `k6 cloud run --local-execution`). Previously, omitting a stack would fall back to the first available stack with a deprecation warning — that fallback has been removed. Likewise, `k6 cloud login` now requires both a token and a stack; passing one without the other fails with an explicit error.

**Migration:** Ensure a stack is always provided via `k6 cloud login --stack` passing the stack slug or the full URL. Alternatively, the `K6_CLOUD_STACK_ID` environment variable, or the `stackID` script option are available to be set before running any `k6 cloud` command .

> ### Removed legacy configuration file path migration [#5609](https://github.com/grafana/k6/pull/5609)

Reviewer 2: Should we appreciate contributions from external contributors? 
There's others like https://github.com/grafana/k6/pull/5815, mentioned below.

I'm happy with a single line at the end, like [here](https://github.com/grafana/k6/blob/c9931b0e53de996d9e9fd0e23320cdbb24f34a05/release%20notes/v1.6.0.md#external-contributors)

Author: good catch - I caught it earlier, but apperantly never ended up checking all the cases and adding tha tahk yous :facepalm:

Author: [ab93b40](https://github.com/grafana/k6/pull/5868/commits/ab93b401edc574a3cbf30af3168e2e2ffc2cf151)

k6 `v2.0.0-rc1` is here 🎉!

This release marks the first release candidate for k6 v2.0.0 — a major version that completes a long-running cleanup of deprecated APIs, old commands, and obsolete configuration options. Like the v1.0.0-rc1 before it, the purpose of this release is to give the community a chance to test the upgrade path, identify any issues, and migrate scripts or workflows affected by breaking changes. If you encounter any problems, please [report them](https://github.com/grafana/k6/issues).

> Here's a glimpse of what's changed in this release:

Reviewer 3: I think the stackID is a pretty significant change that it should be added here too.

### Removed `K6_BINARY_PROVISIONING` environment variable [#5734](https://github.com/grafana/k6/pull/5734)

The `K6_BINARY_PROVISIONING` environment variable, deprecated in v1.2.0, has been removed.

**Migration:** Remove `K6_BINARY_PROVISIONING` from your environment. Auto-extension-resolution is enabled by default; `K6_AUTO_EXTENSION_RESOLUTION` only needs to be set explicitly if you want to disable it.

### Removed `K6_ENABLE_COMMUNITY_EXTENSIONS` environment variable [#5733](https://github.com/grafana/k6/pull/5733)

The `K6_ENABLE_COMMUNITY_EXTENSIONS` environment variable has been removed. The community and cloud extension catalogs were merged server-side, making this flag a no-op since the catalogs were unified.

**Migration:** Remove `K6_ENABLE_COMMUNITY_EXTENSIONS` from your environment. Community extensions are now resolved through the default build service URL automatically.

### Stack is now required for all `k6 cloud` commands [#5833](https://github.com/grafana/k6/pull/5833)

> Providing a stack is now mandatory for all `k6 cloud` commands (`k6 cloud run`, `k6 cloud upload`, `k6 cloud run --local-execution`). Previously, omitting a stack would fall back to the first available stack with a deprecation warning — that fallback has been removed. Likewise, `k6 cloud login` now requires both a token and a stack; passing one without the other fails with an explicit error.

Reviewer 3: Are there docs on how to get the `stackID`? I'm wondering if we should link to that or try to explain how to find the `stackID`.

Reviewer 3: Never mind, i just remembered that you get an option when logging in 👍 Maybe worth mentioning that the user logs in again?

**Migration:** Remove `K6_BINARY_PROVISIONING` from your environment. Auto-extension-resolution is enabled by default; `K6_AUTO_EXTENSION_RESOLUTION` only needs to be set explicitly if you want to disable it.

### Removed `K6_ENABLE_COMMUNITY_EXTENSIONS` environment variable [#5733](https://github.com/grafana/k6/pull/5733)

The `K6_ENABLE_COMMUNITY_EXTENSIONS` environment variable has been removed. The community and cloud extension catalogs were merged server-side, making this flag a no-op since the catalogs were unified.

**Migration:** Remove `K6_ENABLE_COMMUNITY_EXTENSIONS` from your environment. Community extensions are now resolved through the default build service URL automatically.

### Stack is now required for all `k6 cloud` commands [#5833](https://github.com/grafana/k6/pull/5833)

Providing a stack is now mandatory for all `k6 cloud` commands (`k6 cloud run`, `k6 cloud upload`, `k6 cloud run --local-execution`). Previously, omitting a stack would fall back to the first available stack with a deprecation warning — that fallback has been removed. Likewise, `k6 cloud login` now requires both a token and a stack; passing one without the other fails with an explicit error.

> **Migration:** Ensure a stack is always provided via `k6 cloud login --stack` passing the stack slug or the full URL. Alternatively, the `K6_CLOUD_STACK_ID` environment variable, or the `stackID` script option are available to be set before running any `k6 cloud` command .

Author: ```suggestion
**Migration:** Run `k6 cloud login` which will ask you for the stack and setup correctly for the new version of k6. Alternatively, the `K6_CLOUD_STACK_ID` environment variable, or the `stackID` script option are available to be set before running any `k6 cloud` command .
```

The command already leads you to what needs to be done, so you do not need to provide a flag
